ngsi_adapter/src/test/acceptance/commons/logger_utils.py

```diff
@@ -0,0 +1,154 @@
+    :param level: Level (INFO | DEBUG | WARN | ERROR)
+    :return:
+    """
+
+    global logging_level
+    logging_level = logging.ERROR
+    if "info" == level.lower():
+        logging_level = logging.INFO
+    elif "warn" == level.lower():
+        logging_level = logging.WARNING
+    elif "debug" == level.lower():
+        logging_level = logging.DEBUG
+
+
+def get_logger(name):
+    """
+    Creates new logger with the given name
+    :param name: Name of the logger
+    :return: Logger
+    """
+
+    #logging.config.fileConfig("logging.conf")
+    logger = logging.getLogger("testingLogger")
+
+    # if not len(logger.handlers):
+    #     # File handler
+    #     file_hdlr = logging.FileHandler('logs/monitoring_tests.log')
+    #     formatter = logging.Formatter(LOG_FILE_FORMATTER)
+    #     file_hdlr.setFormatter(formatter)
+    #     logger.addHandler(file_hdlr)
+    #     logger.setLevel(logging.DEBUG)
+    #
+    #     # Console hadler
+    #     console_hdlr = logging.StreamHandler()
+    #     formatter = logging.Formatter(LOG_CONSOLE_FORMATTER)
+    #     console_hdlr.setFormatter(formatter)
+    #     logger.addHandler(console_hdlr)
+    #     logger.setLevel(logging_level)
+
+    return logger
+
+
+def _get_pretty_body(headers, body):
+    """
+    Returns a pretty printed body using the Content-Type header information
+    :param headers: Headers for the request/response (dict)
+    :param body: Body to pretty print (string)
+    :return: Body pretty printed (string)
+    """
+
+    if HEADER_CONTENT_TYPE in headers:
+        if HEADER_REPRESENTATION_XML == headers[HEADER_CONTENT_TYPE]:
+            xml_parsed = xml.dom.minidom.parseString(body)
+            pretty_xml_as_string = xml_parsed.toprettyxml()
+            return pretty_xml_as_string
+        else:
+            if HEADER_REPRESENTATION_JSON in headers[HEADER_CONTENT_TYPE]:
+                parsed = json.loads(body)
+                return json.dumps(parsed, sort_keys=True, indent=4)
+            else:
+                return body
+    else:
+        return body
+
+
+def log_print_request(logger, method, url, query_params=None, headers=None, body=None):
+    """
+    Logs an HTTP request data.
+    :param logger: Logger to use
+    :param method: HTTP method
+    :param url: URL
+    :param query_params: Query parameters in the URL
+    :param headers: Headers (dict)
+    :param body: Body (raw body, string)
+    :return: None
+    """
+
+    log_msg = '>>>>>>>>>>>>>>>>>>>>> Request >>>>>>>>>>>>>>>>>>> \n'
+    log_msg += '\t> Method: %s\n' % method
+    log_msg += '\t> Url: %s\n' % url
+    if query_params is not None:
+        log_msg += '\t> Query params: {}\n'.format(str(query_params))
+    if headers is not None:
+        log_msg += '\t> Headers: {}\n'.format(str(headers))
+    if body is not None:
+        log_msg += '\t> Payload sent:\n {}\n'.format(_get_pretty_body(headers, body))
+
+    logger.debug(log_msg)
+
```

Comment:
It is "common" review the test data and process after launching. The objective of these logs is to be human and quickly readable to review it without any other log analysis.

Normally, all messages are shown in the standard output but in this case I have prefered show only the feature execution by Lettuce and the rest of the messages about the test process and dataset have been redirected to the log file.